padertorch/train/hooks.py

```diff
@@ -397,40 +407,27 @@ def __init__(
         self.maximize = maximize
         self.max_checkpoints = max_checkpoints
         self.early_stopping_patience = early_stopping_patience
-        self._json_file = None
         self.ckpt_ranking = []
         self.n_degradations = 0
+        self.last_validation = -1
 
     @property
     def priority(self):
         return Priority.VALIDATION
 
-    @property
-    def json_file(self):
-        if self._json_file is None:
-            raise RuntimeError(
-                'The property json_file will be lazy setted in the pre_step'
-                'function.\n'
-                'The trainer knows, where to store the "validation_state" and '
-                'this hooks sees the trainer first in the pre_step.'
-            )
-        return self._json_file
-
     @property
     def _best_ckpt_name(self):
         return f"ckpt_best_{self.metric}.pth"
 
-    def save_validation_state(self):
-        validation_state = {
+    def state_dict(self):
```

Comment:
shouldn't the state_dict be a property of the hook?

Reply:
Usually yes, but here we want to match pytorch `Module.state_dict()`:
https://github.com/pytorch/pytorch/blob/8195961f20caa18863698d85410d0818bfd67463/torch/nn/modules/module.py#L671
```python
def state_dict(self, destination=None, prefix='', keep_vars=False):
```